src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs

```diff
@@ -193,310 +422,828 @@ static void Main()
+                Diagnostic(ErrorCode.ERR_BadConstType, "S").WithArguments("S").WithLocation(8, 15),
+                // (9,26): error CS0133: The expression being assigned to 'z' must be constant
+                //         const object z = (S)default;
+                Diagnostic(ErrorCode.ERR_NotConstantExpression, "(S)default").WithArguments("z").WithLocation(9, 26)
+                );
+        }
+
+        [Fact]
+        public void ImplicitlyTypedArray()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        var t = new[] { 1, default };
+        System.Console.Write(t[1]);
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.Regular7_1, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "0");
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+
+            var def = nodes.OfType<LiteralExpressionSyntax>().ElementAt(1);
+            Assert.Equal("default", def.ToString());
+            Assert.Equal("System.Int32", model.GetTypeInfo(def).Type.ToTestDisplayString());
+            Assert.Equal("System.Int32", model.GetTypeInfo(def).ConvertedType.ToTestDisplayString());
+            Assert.Null(model.GetSymbolInfo(def).Symbol);
+            Assert.Equal("0", model.GetConstantValue(def).Value.ToString());
+        }
+
+        [Fact]
+        public void CollectionInitializer()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        var t = new System.Collections.Generic.List<int> { 1, default };
+        System.Console.Write($""{t[0]} {t[1]}"");
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.Regular7_1, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "1 0");
+        }
+
+        [Fact]
+        public void MiscDefaultErrors()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        switch (default)
+        {
+            default:
+                break;
+        }
+        lock (default)
+        {
+        }
+        default();
+
+        int i = ++default;
+        var anon = new { Name = default };
+        System.TypedReference tr = __makeref(default);
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.Regular7_1, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics(
+                // (14,17): error CS1031: Type expected
+                //         default();
+                Diagnostic(ErrorCode.ERR_TypeExpected, ")").WithLocation(14, 17),
+                // (6,17): error CS8119: The switch expression must be a value; found default.
```

Comment:
I think the word default in this diagnostic should have some kind of quotes around it. It isn't a word of the sentence, it is referring to syntax in the source.